### PR TITLE
Use DOCtor-RST 1.62.2 and new `no_broken_ref_directive` rule

### PR DESCRIPTION
**Pull request by OskarStark**
Follows
* https://github.com/OskarStark/doctor-rst/pull/1816

cc @MrYamous

### CODE REVIEW

**Comment by OskarStark** [comment]
CleanShot 2024-08-22 at 15 30 29@2x

Build of the image is not done yet: https://github.com/OskarStark/doctor-rst/actions/runs/10509036020/job/29114264490

**Comment by OskarStark** [comment]
@MrYamous looks like we have a lot of false positives

**Comment by MrYamous** [comment]
> @MrYamous looks like we have a lot of false positives

Ooops i didn't except that, i'm on it

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
                     key: ${{ runner.os }}-doctor-rst-${{ steps.extract_base_branch.outputs.branch }}
 
             -   name: "Run DOCtor-RST"
-                uses: docker://oskarstark/doctor-rst:1.61.1
+                uses: docker://oskarstark/doctor-rst:1.62.0
```
Comment (OskarStark) [inline .github/workflows/ci.yaml]:
```suggestion
                uses: docker://oskarstark/doctor-rst:1.62.1
```

**Comment by OskarStark** [comment]
@MrYamous can you please have an additional look? Thanks

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -73,7 +73,7 @@ jobs:
                     key: ${{ runner.os }}-doctor-rst-${{ steps.extract_base_branch.outputs.branch }}
 
             -   name: "Run DOCtor-RST"
-                uses: docker://oskarstark/doctor-rst:1.61.1
+                uses: docker://oskarstark/doctor-rst:1.62.1
```
Comment (OskarStark) [inline .github/workflows/ci.yaml]:
```suggestion
                uses: docker://oskarstark/doctor-rst:1.62.2
```

**Comment by OskarStark** [comment]
Thanks @MrYamous 👍

**Pull request merged**